src/ray/core_worker/fiber.h

```diff
@@ -110,10 +111,23 @@ class FiberState {
                   return;
                 }
               }
-              // The event here is used to make sure fiber_runner_thread_ never
-              // terminates. Because fiber_shutdown_event_ is never notified,
-              // fiber_runner_thread_ will immediately start working on any ready fibers.
-              shutdown_worker_event_.Wait();
+
+              // Boost fiber thread cannot be terminated and joined
```

Comment:
Yes. So the issue is that for asyncio actor, we have two sets of threads to run actor methods: boost fiber threads and python asyncio threads. When a actor method call task comes in, a boost detached fiber will post it to python asyncio thread and wait for it to finish so the fiber can sends the response back to the caller. During shutdown, we simply stop running python coroutines instead of waiting for them to finish (which is impossible for infinite loop actor method), as a result, waiting boost fiber never receives the notification that the corresponding python coroutine finishes so they won't terminate which prevents the fiber thread from being joined.